Doc/library/typing.rst

```diff
@@ -2882,24 +2882,32 @@ Introspection helpers
       Now the annotation is returned unchanged.
 
 .. function:: get_args(tp)
-.. function:: get_origin(tp)
-
-   Provide basic introspection for generic types and special typing forms.
 
-   For a typing object of the form ``X[Y, Z, ...]`` these functions return
-   ``X`` and ``(Y, Z, ...)``. If ``X`` is a generic alias for a builtin or
-   :mod:`collections` class, it gets normalized to the original class.
+   Get type arguments with all substitutions performed: for a typing object
+   of the form ``X[Y, Z, ...]`` return ``(Y, Z, ...)``.
    If ``X`` is a union or :class:`Literal` contained in another
    generic type, the order of ``(Y, Z, ...)`` may be different from the order
    of the original arguments ``[Y, Z, ...]`` due to type caching.
-   For unsupported objects return ``None`` and ``()`` correspondingly.
+   Return ``()`` for unsupported objects.
    Examples::
 
-      assert get_origin(Dict[str, int]) is dict
+      assert get_args(int) == ()
       assert get_args(Dict[int, str]) == (int, str)
+      assert get_args(Union[int, str]) == (int, str)
+
+   .. versionadded:: 3.8
 
+.. function:: get_origin(tp)
+
+   Get the unsubscripted version of a type: for a typing object of the form
+   ``X[Y, Z, ...]`` return ``X``. If ``X`` is a generic alias for a builtin or
+   :mod:`collections` class, it gets normalized to the original class.
```

Comment:
Let's add that for `ParamSpecArgs` and `Kwargs`, it returns the underlying ParamSpec.

Reply:
Done.

Reply:
Thanks!